tests/objects/tillegal_recursion.nim

```diff
@@ -1,5 +1,5 @@
 discard """
-  errormsg: "illegal recursion in type 'object'"
+  errormsg: "inheritance only works with non-final objects"
```

Comment:
Not really, but it's not too bad for now and makes the test green, so it's fine.